perun-core/src/main/java/cz/metacentrum/perun/core/impl/AuthzResolverImpl.java

```diff
@@ -785,7 +807,92 @@ public static PerunPolicy getPerunPolicy(String policyName) throws PolicyNotExis
 		return new ArrayList<>(perunPoliciesContainer.getAllPolicies());
 	}
 
+	/**
+	 * Get RoleManagementRules for the role name from the PerunPoliciesContainer
+	 *
+	 * @param roleName for which will be the rules fetched
+	 * @return RoleManagementRules for the role name
+	 * @throws PolicyNotExistsException of there are no rules for the role name
+	 */
 	public static RoleManagementRules getRoleManagementRules(String roleName) throws RoleManagementRulesNotExistsException {
 		return perunPoliciesContainer.getRoleManagementRules(roleName);
 	}
+
+	@Override
+	public Integer getRoleId(String role) {
+		try {
+			return jdbc.queryForInt("SELECT id FROM roles WHERE name=?", role.toLowerCase());
+		} catch (RuntimeException e) {
+			throw new InternalErrorException(e);
+		}
+	}
+
+	@Override
+	public void setRole(PerunSession sess, Map<String, Integer> mappingOfValues, String role) throws RoleAlreadySetException {
+		String query = prepareQueryToSetRole(mappingOfValues);
+
+		try {
+			jdbc.update(query);
+		} catch (DataIntegrityViolationException e) {
+			throw new RoleAlreadySetException(role);
+		} catch (RuntimeException e) {
+			throw new InternalErrorException(e);
+		}
+	}
+
+	@Override
+	public void unsetRole(PerunSession sess, Map<String, Integer> mappingOfValues, String role) throws RoleNotSetException {
+		String query = prepareQueryToUnsetRole(mappingOfValues);
+
+		try {
+			if (0 == jdbc.update(query)) {
+				throw new RoleNotSetException(role);
+			}
+		} catch (RuntimeException e) {
+			throw new InternalErrorException(e);
+		}
+	}
+
+	/**
+	 * Create query to set role according to the mapping of values
+	 *
+	 * @param mappingOfValues from which will be the query created
+	 * @return sql query
+	 */
+	private String prepareQueryToSetRole(Map<String, Integer> mappingOfValues) {
+		String columnsFromMapping;
+		String valuesFromMapping;
+		List<String> columnNames = new ArrayList<>();
+		List<String> columnValues = new ArrayList<>();
+
+		for (String columnName: mappingOfValues.keySet()) {
+			columnNames.add(columnName);
+			columnValues.add(mappingOfValues.get(columnName).toString());
+		}
+
+		columnsFromMapping = StringUtils.join(columnNames, ",");
+		valuesFromMapping = StringUtils.join(columnValues, ",");
+
+		return "insert into authz (" + columnsFromMapping + ") values (" + valuesFromMapping + ")";
```

Comment:
I am a little worried about these two mapping values. You should at least escape them to prevent some sort of SQL injection even if it looks like you should have these variables under control because they are coming from configuration files.

Reply:
Same for method prepareQueryToUnsetRole.

Reply:
It is handled in the method createMappingOfValues. There is a pattern that allows using only letters, numbers, and underscores.

Reply:
Yes, but the problem is if anyone will use this method separately. You should move this pattern right into this method.

Reply:
I have moved the check to the Impl layer, where the query is created.